sql/core/src/main/scala/org/apache/spark/sql/execution/command/AnalyzeColumnCommand.scala

```diff
@@ -89,6 +97,46 @@ case class AnalyzeColumnCommand(
     columnsToAnalyze
   }
 
+  private def analyzeColumnInCatalog(sparkSession: SparkSession): Unit = {
+    val sessionState = sparkSession.sessionState
+    val tableMeta = sessionState.catalog.getTableMetadata(tableIdent)
+    if (tableMeta.tableType == CatalogTableType.VIEW) {
+      // Analyzes a catalog view if the view is cached
+      val cacheManager = sparkSession.sharedState.cacheManager
+      val plan = sparkSession.table(tableIdent.quotedString).logicalPlan
+      cacheManager.lookupCachedData(plan) match {
+        case Some(cachedData) =>
+          val columnsToAnalyze = getColumnsToAnalyze(
+            tableIdent, cachedData.plan, columnNames, allColumns)
+          cacheManager.analyzeColumnCacheQuery(sparkSession, cachedData, columnsToAnalyze)
+        case _ =>
+          throw new AnalysisException("ANALYZE TABLE is not supported on views.")
```

Comment:
This can be `ANALYZE TABLE is not supported on views`, but I'm fine with the current one too because it's too specific case.